src/Core/Module/ModuleRepository.php

```diff
@@ -231,19 +280,25 @@ private function getModuleDatabaseAttributes(string $moduleName): array
     }
 
     /**
-     * @param Module[] $modules
-     *
-     * @return Module[]
+     * @return array
      */
-    private function mergeWithModulesFromHook(array $modules): array
+    private function getHookModulesForActionListModules()
     {
         $actionListModules = $this->hookManager->exec('actionListModules', [], null, true);
         $externalModules = array_values($actionListModules ?? []);
-        if (empty(reset($externalModules))) {
-            return $modules;
-        }
 
-        foreach (array_merge(...$externalModules) as $externalModule) {
+        return empty($externalModules) ? $externalModules : array_merge(...$externalModules);
```

Comment:
I think you can do that directly?
```suggestion
        return array_merge(...$externalModules);
```

Reply:
I thought so but CI crashed with empty external modules due to empty `$externalModules`

It looks [fixed in php 7.4](https://www.php.net/manual/en/function.array-merge.php)